### PR TITLE
Bulk Publishing: Follow Ups - Match order of metrics list in Data Entry and Publish Review pages

### LINKED ISSUE #590: [Frontend] Bulk Publishing: Follow Ups - Match order of metrics list in Data Entry and Publish Review pages

### PR DESCRIPTION
**Pull request by mxosman**
## Description of the change

Sorts the list of metrics on the publish review page to match the order of the metrics list in the data entry page. 
[Thank you Michelle & Lili for unblocking me w/ your great suggestions!]

https://user-images.githubusercontent.com/59492998/234736512-7533099b-e1dd-4cbe-bd64-944377b3fbc7.mov

## Related issues

Closes #590

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: publisher/src/components/Reports/DataEntryReview.tsx [hunk 1]
```diff
@@ -139,18 +141,25 @@ const DataEntryReview = () => {
...
+            datapoints: datapointsByMetric[metric.key],
+            display_name: metric.displayName,
+            key: metric.key,
+            metricHasError: checkMetricForErrors(metric.key),
+            metricHasValidInput: Boolean(
+              formStore.metricsValues?.[reportID]?.[metric.key]?.value
+            ),
+          };
+          return [...acc, reviewMetric];
+        }, [] as ReviewMetric[])
```
Comment (mxosman) [inline publisher/src/components/Reports/DataEntryReview.tsx:156]: No changes in L144-156

**Review by morden35 — APPROVED** [review]
amazing! and done in so few lines of code! bravo

**Review by lilidworkin — APPROVED** [review]
Looks perfect!

**Comment by mxosman** [comment]
Thank you both so much!!

**Pull request merged**